cloud-sql-proxy/README.MD

```diff
@@ -64,6 +64,80 @@ gcloud dataproc clusters create <CLUSTER_NAME> \
     --metadata "additional-cloud-sql-instances=<PROJECT_ID>:<REGION>:<ANOTHER_INSTANCE_NAME>"
 ```
 
+## Private IP Clusters and Cloud SQL Instances
+
+Connecting to a cloud sql instance from a private IP cluster requires some additional setup.
+
+1. Create a network.
+
+First, create a network:
+```
+gcloud compute networks create <NETWORK>
+```
+Note that regional subnets will be created automatically.
+To list these subnets:
+```
+gcloud compute networks subnets list --network <NETWORK>
+```
+
+2. Enable private IP for a subnet.
+
+Choose the subnet that is in the region you want to use.
+Enable private ip google access for your subnet:
+```
+gcloud compute networks subnets update <SUBNET> \
+    --region <REGION> \
+    --enable-private-ip-google-access
+```
+
+3. Create a cloud sql instance
+
+Create a second generation cloud sql instance:
+
+```
+gcloud sql instances create <INSTANCE_NAME> \
+    --database-version="MYSQL_5_7" \
+    --activation-policy=ALWAYS \
+    --gce-zone <ZONE>
+```
+Make sure that the zone is in the region you chose earlier.
+
+4. Enable the service networking API
+```
+gcloud services enable servicenetworking.googleapis.com
+```
+
+5. Configure the cloud sql instance to use private IP
+Configure your cloud sql instance to use private IP by following [these instructions](https://cloud.google.com/sql/docs/mysql/configure-private-ip#configuring_an_existing_instance_to_use_private_ip).
+Important notes:
+When selecting a network, make sure to select the same one that you created earlier.
+Make sure to click Save after making changes to your cloud sql instance and note that the update may take a while.
+You can confirm that the changes were made by describing your cloud sql instance:
+```
+gcloud sql instances describe <INSTANCE_NAME>
+```
+If both a public and a private IP appear under ipAddresses, then you have successfully completed this step.
+
+6. Create a cluster using this initialization action
+
+To create a private IP cluster that points to your cloud sql instance, do the following:
+
+```
+gcloud dataproc clusters create <CLUSTER_NAME> \
+    --scopes sql-admin \
+    --initialization-actions gs://dataproc-initialization-actions/cloud-sql-proxy.sh \
```

Comment:
path should be: gs://dataproc-initialization-actions/cloud-sql-proxy/cloud-sql-proxy.sh

Reply:
Nice catch - done